src/m365/commands/login.ts

```diff
@@ -204,39 +201,14 @@ class LoginCommand extends Command {
         throw new CommandError(error.message);
       }
 
-      if (this.debug) {
-        await logger.log({
-          connectedAs: accessToken.getUserNameFromAccessToken(auth.service.accessTokens[auth.defaultResource].accessToken),
-          authType: AuthType[auth.service.authType],
-          appId: auth.service.appId,
-          appTenant: auth.service.tenant,
-          accessToken: JSON.stringify(auth.service.accessTokens, null, 2),
-          cloudType: CloudType[auth.service.cloudType]
-        });
-      }
-      else {
-        await logger.log({
-          connectedAs: accessToken.getUserNameFromAccessToken(auth.service.accessTokens[auth.defaultResource].accessToken),
-          authType: AuthType[auth.service.authType],
-          appId: auth.service.appId,
-          appTenant: auth.service.tenant,
-          cloudType: CloudType[auth.service.cloudType]
-        });
-      }
+      await logger.log({
+        ...auth.getIdentityDetails(auth.service, this.debug),
+        connectedAs: auth.service.identityName // (Deprecated), added for backwards compatibility
+      });
     };
 
-    try {
-      await auth.clearConnectionInfo();
-    }
-    catch (error: any) {
-      if (this.debug) {
-        await logger.logToStderr(new CommandError(error));
-      }
-    }
-    finally {
-      logout();
-      await login();
-    }
+    auth.service.deactivateIdentity();
```

Comment:
The entire section was not needed in my new setup... or so I'm thinking.

Reply:
Why not?

Reply:
Well, for starters, the try catch is wrapping the clearConnectionInfo call. Which I have removed because we don't need to clear the identity and msal caches before logging in now.

Reply:
I wonder though if the same doesn't apply to deactivateIdentity: what if it fails? Should the whole login fail?

Reply:
deactivateIdentity currently doesn't really do something aside from resetting properties. 
We could of course wrap it anyway, or make it persist the deactivation...